src/cdk/testing/using-component-harnesses.md

```diff
@@ -0,0 +1,264 @@
+  const okButton = await loader.getHarness(MatButtonHarness.with({selector: '.confirm'});
+  expect(fixture.componentInstance.confirmed).toBe(false);
+  expect(await okButton.isDisabled()).toBe(false);
+  await okButton.click();
+  expect(fixture.componentInstance.confirmed).toBe(true);
+});
+```
+
+Note that the code above has does not call `fixture.detectChanges()`, something you commonly see in
+unit tests. The CDK's component harnesses automatically invoke change detection after performing
+actions and before reading state. The harness also automatically waits for the fixture to be stable,
+which will cause the test to wait for `setTimeout`, `Promise`, etc.
+
+## Comparison with and without component harnesses
+
+Consider an `<issue-report-selector>` component that you want to test. It allows a user to
+choose an issue type and display the necessary form create report for that issue type. You need a 
+test to verify that when the user chooses an issue type the proper report displays. First consider
+what the test might look like without using component harnesses:
+
+```ts
+describe('issue-report-selector', () => {
+  let fixture: ComponentFixture<IssueReportSelector>;
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      imports: [IssueReportSelectorModule],
+      declarations: [IssueReportSelector],
+    }).compileComponents();
+
+    fixture = TestBed.createComponent(IssueReportSelector);
+    fixture.detectChanges();
+  });
+
+  it('should switch to bug report template', async () => {
+    expect(fixture.debugElement.query('bug-report-form')).toBeNull();
+    const selectTrigger = fixture.debugElement.query(By.css('.mat-select-trigger'));
+    selectTrigger.triggerEventHandler('click', {});
+    fixture.detectChanges();
+    await fixture.whenStable();
+    const options = document.querySelectorAll('.mat-select-panel mat-option');
+    options[1].click(); // Click the second option, "Bug".
+    fixture.detectChanges();
+    await fixture.whenStable();
+    expect(fixture.debugElement.query('bug-report-form')).not.toBeNull();
+  });
+});
+```
+
+The same test, using the Angular Material component harnesses might look like the following:
+
+```ts
+describe('issue-report-selector', () => {
+  let fixture: ComponentFixture<IssueReportSelector>;
+  let loader: HarnessLoader;
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      imports: [IssueReportSelectorModule],
+      declarations: [IssueReportSelector],
+    }).compileComponents();
+
+    fixture = TestBed.createComponent(IssueReportSelector);
+    fixture.detectChanges();
+    loader = TestbedHarnessEnvironment.loader(fixture);
+  });
+
+  it('should switch to bug report template', async () => {
+    expect(fixture.debugElement.query('bug-report-form')).toBeNull();
+    const select = await loader.getHarness(MatSelect);
+    await select.open();
+    const bugOption = await select.getOption({text: 'Bug'});
```

Comment:
The select harness doesn't have a method like this now. Add it? We should add a `selectMatchingOption` method to the select harness to make this even more concise, _maybe_ also something like `openAndSelectMatchingOption`

Reply:
I think @crisbeto is working on something like this in an open PR somewhere. But yeah, I do want to take a pass over all the harnesses before 9.0 and make sure they have the methods people need to do common actions quickly and easily